### PR TITLE
Added credits for DataSources and Model

### PR DESCRIPTION
**Pull request by tfili**
- Added a `credit` options for `CZML`, `KML` and `GeoJson` data sources as well as for `Model`.
- Added ion credits to those classes as well that can't be removed from the instance.
- We now allow `scene.postRender` callbacks to add credits. This was required for `DataSource` credits.
- Added tests for all the classes.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @tfili!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/datasource-model-credits/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by tfili** [comment]
@lilleyse I just wanted to make sure what I did in `Scene` is ok. I don't see any ill effects from it.

File: Source/DataSources/CzmlDataSource.js [hunk 1]
```diff
@@ -2150,11 +2152,28 @@ define([
...
             promise = czml.fetchJson();
             sourceUri = defaultValue(sourceUri, czml.clone());
+
+            // Add resource credits to our list of credits to display
+            var resourceCredits = dataSource._resourceCredits;
+            var credits = czml.credits;
+            if (defined(credits)) {
+                var length = credits.length;
+                for (var i = 0; i < length; i++) {
+                    resourceCredits.push(credits[i]);
```
Comment (hpinkos) [inline Source/DataSources/CzmlDataSource.js]:
```suggestion
                    resourceCredits.push(new Credit(credits[i)]);
```

These definitely won't be `Credit` instances, right?

**Comment by hpinkos** [comment]
I think hooking this in in `Viewer` is too high level.  `DataSourceDisplay` requires a `Scene`, right? I would add a new `beginFrame` event to `Scene` that `DataSourceDisplay` listens to.  This event would fire sometime between when `creditDisplay.beginFrame` and `creditDisplay.endFrame` are called in `Scene.render`.  `DataSourceDisplay` would listen to that event and call the `updateCredits` function when that is raised.  Does that make sense?

File: Source/DataSources/CzmlDataSource.js (hunk 1 above)
Comment (tfili) [reply Source/DataSources/CzmlDataSource.js]: They definitely are `Credit` objects. We iterate the `endpoint.attributions` and call `Credit.getIonCredit`, which returns a `Credit` instance.

Comment (tfili) [reply Source/DataSources/CzmlDataSource.js]: See `IonResource.getCreditsFromEndpoint`

**Comment by tfili** [comment]
> I think hooking this in in Viewer is too high level. DataSourceDisplay requires a Scene, right?

Agreed.

> I would add a new beginFrame event to Scene that DataSourceDisplay listens to.

Is there a need for another event? There is already `[begin|end]Update` and a `[pre|post]Render` events. Is there any ill effects from moving them out to where I have them (moved out of local function `render` into `Scene.render`).

**Comment by hpinkos** [comment]
@tfili yeah, looking at the code, you'll either need to add another event, or move where `creditDisplay.startFrame` gets called.

**Comment by tfili** [comment]
This should be good.
- Start cesium server
- Run ion locally as a beta user
- Add a Model, CZML, GeoJSON and a KML and give them all attributions
- Click the View in Sandcastle link for each of them
- Verify you see your attributions

**Comment by hpinkos** [comment]
CHANGES.md?

**Comment by hpinkos** [comment]
Works great @tfili!  Update CHANGES and I'll merge this in

**Comment by tfili** [comment]
@hpinkos Done!

**Pull request merged**